run_ci.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Exit immediately if a command exits with a non-zero status
+set -e
+
+# The path to the YAML file that defines the CI workflows
+YAML_FILE=".github/workflows/lint-and-test.yml"
+
+# Check if the YAML file exists
+if [ ! -f "$YAML_FILE" ]; then
+    echo "YAML file $YAML_FILE does not exist."
+    exit 1
+fi
+
+# Extract all lines that contain 'cargo' commands from the YAML file, 
+# excluding ones with '--ignored' or 'test'
+cargo_commands=$(grep -E '^\s*run:.*cargo' "$YAML_FILE" | grep -v -- '--ignored' | grep -v 'test' | sed -E 's/^\s*run:\s*//')
```

Comment:
This is the output I get:
```bash
curl https://sh.rustup.rs -sSf | bash -s -- -y --profile minimal && source ~/.cargo/env
cargo check --no-default-features
cargo check --all-targets
cargo check --all-targets --all-features
cargo check -p proof-of-sql --no-default-features
cargo check -p proof-of-sql --all-features
cargo check -p proof-of-sql --no-default-features --features="blitzar"
cargo check -p proof-of-sql --no-default-features --features="arrow"
cargo check -p proof-of-sql --no-default-features --features="rayon"
cargo check -p proof-of-sql --no-default-features --features="perf"
cargo check -p proof-of-sql --no-default-features --features="std"
curl https://sh.rustup.rs -sSf | bash -s -- -y --profile minimal && source ~/.cargo/env
cargo clippy --all-targets --all-features -- -D warnings
cargo fmt --all -- --config imports_granularity=Crate,group_imports=One --check
cargo +nightly udeps --all-targets
```

* This isn't the full CI since it's missing tests. So, we should rename this file to be more descriptive.
    * I see now that the last line says this, but it wasn't obvious to me immediately.
* I don't think we want the `rustup` commands here.